tests/test_lines.py

```diff
@@ -25,8 +28,80 @@ def test_lines(example_model_and_crs):
 
    # Check fault can be loaded in again
    model.store_epc()
+   model = rq.Model(epc_file = model.epc_file)
+   reload = resqpy.lines.Polyline(parent_model = model, uuid = line.uuid)
+   assert reload.citation_title == title
+
    fault_interp = resqpy.organize.FaultInterpretation(model, uuid = line.rep_int_uuid)
    fault_feature = resqpy.organize.TectonicBoundaryFeature(model, uuid = fault_interp.feature_uuid)
 
    # Check title matches expected title
    assert fault_feature.feature_name == title
+
+
+def test_lineset(example_model_and_crs):
+
+   # Set up a PolylineSet
+   title = 'Nazcas'
+   model, crs = example_model_and_crs
+   line1 = resqpy.lines.Polyline(parent_model = model,
+                                 title = title,
+                                 set_crs = crs.uuid,
+                                 set_crsroot = crs.crs_root,
+                                 set_bool = True,
+                                 set_coord = np.array([[0, 0, 0], [1, 1, 1]]))
+
+   line2 = resqpy.lines.Polyline(parent_model = model,
+                                 title = title,
+                                 set_crs = crs.uuid,
+                                 set_crsroot = crs.crs_root,
+                                 set_bool = True,
+                                 set_coord = np.array([[0, 0, 0], [2, 2, 2]]))
+
+   lines = resqpy.lines.PolylineSet(parent_model = model, title = title, polylines = [line1, line2])
+
+   lines.write_hdf5()
+   lines.create_xml()
+
+   # Check lines can be loaded in again
+   model.store_epc()
+   model = rq.Model(epc_file = model.epc_file)
+   reload = resqpy.lines.PolylineSet(parent_model = model, uuid = lines.uuid)
+   assert len(reload.polys) == 2, f'Expected two polylines in the polylineset, found {len(reload.polys)}'
+   assert (reload.count_perpol == [2, 2]).all(), f'Expected count per polyline to be [2,2], found {reload.count_perpol}'
+
+
+def test_charisma(example_model_and_crs, example_data_path):
+   # Set up a PolylineSet
+   model, crs = example_model_and_crs
+   charisma_file = os.path.join(example_data_path, "Charisma_example")
+   lines = resqpy.lines.PolylineSet(parent_model = model, charisma_file = charisma_file)
+   lines.write_hdf5()
+   lines.create_xml()
+
+   model.store_epc()
+   model = rq.Model(epc_file = model.epc_file)
+   reload = resqpy.lines.PolylineSet(parent_model = model, uuid = lines.uuid)
+
+   assert reload.title == 'Charisma_example'
+   assert (reload.count_perpol == [
+      4, 5, 4, 5, 5
+   ]).all(), f"Expected count per polyline to be [4,5,4,5,5], found {reload.count_perpol}"
+   assert len(reload.coordinates) == 23, f"Expected length of coordinates to be 23, found {len(reload.coordinates)}"
+
+
+def test_irap(example_model_and_crs, example_data_path):
+   # Set up a PolylineSet
+   model, crs = example_model_and_crs
+   irap_file = os.path.join(example_data_path, "IRAP_example")
```

Comment:
A minor comment: when using Pathlib objects rather than strings there is a neat high-level syntax for path operations. One can simply write `irap_file = example_data_path / "IRAP_example"`

P.S., should those files have a file extension, like ".txt"?

Reply:
Oddly enough I dislike the overloading of / for path concatenation. However, listen to @connortann , not me on this one!

Reply:
Either is fine I think.

However - we should probably make sure that all user-facing resqpy code that accepts filepaths is able to handle `pathlib.Path` objects as well as simple strings. `pathlib` (the higher-level builtin python lib for file paths) is quite widely used these days rather than `os.path`.

Unless there are objections, I'll open a separate issue that that one and work on it.